### ISSUE TITLE
[MDC-103] <No mixin named mdc-typography-baseline-top>

### ISSUE THREAD

**Issue by zdolson** [material-components/material-components-web-codelabs#52]
Following tutorial MDC-103 Web: Material Theming with Color, Shape, Elevation and Type (Web) (https://codelabs.developers.google.com/codelabs/mdc-103-web/#3)

On part 4, we introduce the typography module. After adding the snippets of code for *app.scss* and *home.html*, refreshing the localhost:8080 page results in no changes due to webpack throwing a compilation error.
```
ERROR in ./app.scss
    Module build failed: 
      @include mdc-typography-baseline-top($mdc-list-item-primary-text-baseline-height_);
              ^
          No mixin named mdc-typography-baseline-top
          in node_modules/@material/list/mdc-list.scss (line 109, column 12)
     @ multi (webpack)-dev-server/client?http://localhost:8080 ./app.scss
```

After looking through some of the typography module code, perhaps there is confusion between the imported *_mixins.scss* file and *@material/typography/mixins*?

Issue from the tutorial side is line 113 in *app.scss*:
```
@include mdc-list-item-primary-text-ink-color(on-primary);
```
This line is then calling line 109 (line 3 in the following class snippet) from *node_modules/@material/list/mdc-list.scss*:
```
.mdc-list-item__primary-text {
  @include mdc-typography-overflow-ellipsis;
  @include mdc-typography-baseline-top($mdc-list-item-primary-text-baseline-height_);
  @include mdc-typography-baseline-bottom($mdc-list-item-secondary-text-baseline-height_);

  display: block;

  // stylelint-disable plugin/selector-bem-pattern
  .mdc-list--dense & {
    @include mdc-typography-baseline-top($mdc-list-dense-item-primary-text-baseline-height_);
    @include mdc-typography-baseline-bottom($mdc-list-item-secondary-text-baseline-height_);
  }
  // stylelint-enable plugin/selector-bem-pattern
}
```

Hope this helps address the issue properly.

**Comment by acdvorak** [comment]
Hi there!

When you ran `npm install @material/typography@0.35.0`, did you include the `@0.35.0` at the end?

Did you manually `npm install` any other `@material` packages without specifying a version number?

Are you using npm 4 or earlier? (Support for `package-lock.json` was added in npm 5.)

Thanks!

**Comment by zdolson** [comment]
Hey!

Yeah I installed the package with the specific version requirement, and when it still wasn't working, I tried it without the requirement and still had issues with it.

I believe I installed the other packages exactly as specified in the tutorials leading up to this point. Also when I ran into this issue, I tried installing all the ```@material``` packages as specified as an optional step in the MDC-101.

To clarify, the order in which I performed these additional downloads was:
Strictly following tutorial -> no ```@0.35.0``` requirement -> all ```@material``` packages

My current version of npm is 5.6.0.

**Comment by kfranqueiro** [comment]
We were discussing this again today, and realized that while the 103 codelab pins its install commands at 0.35.0, the instructions in 101 and 102 were not doing the same, which could potentially lead to mixed results and errors when following each of these codelabs consecutively. The commands in MDC-101 and MDC-102 have since been corrected (thanks @williamernest!).

If you clear your `node_modules` and `package.json` dependencies and then re-do the `npm install` commands for each of the packages installed throughout 101, 102, and 103, all with `@0.35.0`, it should work. Alternatively, starting with `mdc-103/starter` should already have the previous packages from 101 and 102 fixed at v0.35.0.

Hope this helps, and sorry for all the trouble! I'll also be entering an issue for us to update the codelabs to use a more recent MDC Web release which should also avoid this particular issue.

**Comment by kfranqueiro** [comment]
I'm going to close this since we've resolved it in the contents of the codelabs.
